DEV-1440 Change Flask to FastAPI

spacid opened the pull request:
The incoming premis events were processed synchronously. This process can take up to more than 10 seconds per event, resulting in a timeout from MediaHaven. Flask has been switched to FastAPI in order to handle these requests in a asynchronous way. The uWSGI server has been swapped out with an ASGI implementation, namely Uvicorn.

The correlation functionality provided by the `viaa-chassis` has been removed for now as there is a dependency on Flask.

Furthermore, the status code of a successful response is of handle_event  is changed from 200 to 202 as per docs:

> For example, let's say you receive a file that must go through a slow process, you can return a response of "Accepted" (HTTP 202) and process it in the background.

https://fastapi.tiangolo.com/pt/tutorial/background-tasks/

Lastly, the response of handle_event is now JSON and a bit more descriptive.

spacid committed:
DEV-1440 Change Flask with FastAPI

The incoming premis events are processed synchronously. This process can
take up to more than 10 seconds per event, resulting in a timeout from MH.
Switch to FastAPI in order to handle these requests in a asynchronous way.

Correlation has been removed for now as there is a dependecy on Flask.

spacid committed:
DEV-1440 Swap uWSGI with Uvicorn

We need an ASGI implementation instead of a WSGI one. So, uWSGI is swapped
out for Uvicorn.

spacid committed:
DEV-1440 Rebuild requirements.txt

The requirements.txt has been rebuild via `pip freeze`, starting from the
necessary packages.

spacid committed:
DEV-1440 Adapt Dockerfile

- Change the CMD so it points to the new startpoint.
- Remove installing gcc and libc6-dev as we don't compile uWSGI anymore.

spacid committed:
DEV-1440 Adapt README to reflect new state

spacid committed:
DEV-1440 Change status code of successful response

Change status code of successful reponse of handle_event from 200 to 202
as per docs:

"For example, let's say you receive a file that must go through a slow
process, you can return a response of "Accepted" (HTTP 202) and process
it in the background."

https://fastapi.tiangolo.com/pt/tutorial/background-tasks/

spacid committed:
DEV-1440 Change response to return JSON

Make the response of handle_event a bit more descriptive.

ghost commented:
## DeepCode failed to analyze this pull request
Something went wrong despite trying multiple times, sorry about that.
Please comment this pull request with "Retry DeepCode" to manually retry, or [contact us](https://www.deepcode.ai/feedback?select=4&utm_source=gh_review) so that a human can look into the issue.

maartends reviewed (commented):
Nice! :+1:

spacid committed:
DEV-1440 Make Uvicorn logging more uniform

Messages logged by Uvicorn are actually logged twice: once by the Uvicorn
logger and once by the logger of the application. The format of the Uvicorn
loggers is also not in JSON format, so remove the handlers of those loggers.

Disable the access logs as logging of the health check pollutes the logs
too much.

Set the logging level of Uvicorn to the level defined in the config.yml.

maartends reviewed (approved):
QAS-test! :+1:  :upside_down_face:
